drivers/i2c/i2c_mchp_xec.c

```diff
@@ -240,13 +307,65 @@ static int wait_completion(const struct device *dev)
 		return -EBUSY;
 	}
 
+	/* Check if Slave responded with ACK/NACK */
+	if (MCHP_I2C_SMB_STS_RO(ba) & MCHP_I2C_SMB_STS_LRB_AD0) {
+		return -EIO;
+	}
+
 	return 0;
 }
 
-static bool check_lines(uint32_t ba)
+static int send_stop_and_wait(const struct device *dev)
 {
-	return ((!(MCHP_I2C_SMB_BB_CTRL(ba) & MCHP_I2C_SMB_BB_CLKI_RO)) ||
-		(!(MCHP_I2C_SMB_BB_CTRL(ba) & MCHP_I2C_SMB_BB_DATI_RO)));
+	const struct i2c_xec_config *config =
+		(const struct i2c_xec_config *const) (dev->config);
+	uint32_t ba = config->base_addr;
+
+	MCHP_I2C_SMB_CTRL_WO(ba) = (MCHP_I2C_SMB_CTRL_PIN |
+				    MCHP_I2C_SMB_CTRL_ESO |
+				    MCHP_I2C_SMB_CTRL_STO |
+				    MCHP_I2C_SMB_CTRL_ACK);
+
+	clear_data_flags(dev);
+
+	return wait_bus_free(dev);
```

Comment:
send_stop_and_wait returns a value but this value is not being used anywhere in the code (except in one place).

Reply:
My policy is to return the original error to the caller not a timeout from attempting to generate STOP in response to the first error.  This controller requires you to wait for it to recognize the bus is no longer in use (busy) after a STOP is generated.  It is not safe to trigger STOP generation and then return saying everything is OK. Also, in certain error scenarios the controller should be told to transmit STOP to let the external device know the transfer is over.

Reply:
Agree to all that. However, i am simply pointing out that we got to use the return value sent by send_stop_and_wait but it is not being used.